### PR TITLE
Made a few tweaks to pipelines

### LINKED ISSUE #27157: Default the pipe scheduler to the thread pool instead of inline
As a safe default for the 99% of people we should use the ThreadPool PipeScheduler instead of the Inline one. Running inline can cause unintended consequences (like deadlocks).

### PR DESCRIPTION
**Pull request by davidfowl**
- Renamed the file containing the class IDuplexPipe
- Changed the default scheduler to use the ThreadPool if none was specified

Resolves #27157

### CODE REVIEW

File: src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs [hunk 1]
```diff
@@ -94,8 +94,8 @@ public Pipe(PipeOptions options)
             _minimumSegmentSize = options.MinimumSegmentSize;
             _pauseWriterThreshold = options.PauseWriterThreshold;
             _resumeWriterThreshold = options.ResumeWriterThreshold;
-            _readerScheduler = options.ReaderScheduler ?? PipeScheduler.Inline;
-            _writerScheduler = options.WriterScheduler ?? PipeScheduler.Inline;
+            _readerScheduler = options.ReaderScheduler ?? PipeScheduler.ThreadPool;
+            _writerScheduler = options.WriterScheduler ?? PipeScheduler.ThreadPool;
```
Comment (stephentoub) [inline src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs]: Thank you.

File: src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs [hunk 2]
```diff
@@ -94,8 +94,8 @@ public Pipe(PipeOptions options)
             _minimumSegmentSize = options.MinimumSegmentSize;
             _pauseWriterThreshold = options.PauseWriterThreshold;
             _resumeWriterThreshold = options.ResumeWriterThreshold;
-            _readerScheduler = options.ReaderScheduler ?? PipeScheduler.Inline;
-            _writerScheduler = options.WriterScheduler ?? PipeScheduler.Inline;
+            _readerScheduler = options.ReaderScheduler ?? PipeScheduler.ThreadPool;
```
Comment (pakrym) [inline src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs]: We either need to move it to PipeOptions or move defaults for memory pool here, it's a bit all over the place right now.

**Comment by davidfowl** [comment]
> Unify pipe defaults and LGTM

What do you mean here? Do you want the pipe defaults to specify everything and have less defaults that are overwritten in the `Pipe` itself?

**Comment by pakrym** [comment]
We have default for memory pool set PipeOptions:

https://github.com/dotnet/corefx/blob/820949492d84ed81c6f793e142b54f33cbd65524/src/System.IO.Pipelines/src/System/IO/Pipelines/PipeOptions.cs#L40

And for schedulers in Pipe. I'd like to move them all to one place.

**Comment by davidfowl** [comment]
Got it.

**Comment by davidfowl** [comment]
Failures are unrelated

**Pull request merged**